compositor_render/src/renderer.rs

```diff
@@ -9,15 +18,32 @@ pub enum RendererNewError {
     FailedToInitWgpuCtx(#[from] WgpuCtxNewError),
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum RendererRegisterTransformationError {
+    #[error("failed to register a transformation in the transformation registry")]
+    TransformationRegistryError(#[from] registry::RegisterError),
+}
+
 impl Renderer {
     pub fn new() -> Result<Self, RendererNewError> {
         Ok(Self {
-            wgpu_ctx: WgpuCtx::new()?,
+            wgpu_ctx: Rc::new(WgpuCtx::new()?),
+            registry: TransformationRegistry::new(),
         })
     }
+
+    pub fn register_transformation<T: Transformation>(
+        &mut self,
+        provider: fn(Rc<WgpuCtx>) -> T,
+    ) -> Result<(), RendererRegisterTransformationError> {
+        self.registry
+            .register(Box::new(provider(self.wgpu_ctx.clone())))?;
+
+        Ok(())
+    }
 }
 
-struct WgpuCtx {
+pub struct WgpuCtx {
     #[allow(dead_code)]
     device: wgpu::Device,
```

Comment:
I think the fields have to be public as well

Reply:
true